sqlite-cloud/tutorial-geopoly.mdx

```diff
@@ -0,0 +1,834 @@
+      center: [lng, lat],
+      zoom,
+    });
+
+    const geocoder = new MapboxGeocoder({
+      accessToken: mapboxgl.accessToken,
+      mapboxgl,
+      zoom: 12,
+    });
+
+    const fullscreenCtrl = new mapboxgl.FullscreenControl({
+      container: mapContainerRef.current,
+    });
+
+    const geolocateCtrl = new mapboxgl.GeolocateControl({
+      fitBoundsOptions: {
+        maxZoom: 12,
+      },
+      positionOptions: {
+        enableHighAccuracy: true,
+      },
+      trackUserLocation: true,
+    });
+
+    mapRef.current.addControl(geocoder);
+    mapRef.current.addControl(fullscreenCtrl);
+    mapRef.current.addControl(geolocateCtrl);
+
+    function updateCoordinates() {
+      const { lng, lat } = mapRef.current.getCenter();
+      setLng(lng.toFixed(4));
+      setLat(lat.toFixed(4));
+      setZoom(mapRef.current.getZoom().toFixed(2));
+    }
+
+    mapRef.current.on('move', updateCoordinates);
+
+    geocoder.on('result', (e) => {
+      const existingMarkers = document.getElementsByClassName('marker');
+      while (existingMarkers[0]) {
+        existingMarkers[0].remove();
+      }
+
+      const popUps = document.getElementsByClassName('mapboxgl-popup');
+      while (popUps[0]) {
+        popUps[0].remove();
+      }
+
+      const [lng, lat] = e.result.geometry.coordinates;
+      queryGeopoly(lng, lat);
+    });
+
+    return () => {
+      mapRef.current.removeControl(geocoder);
+      mapRef.current.removeControl(fullscreenCtrl);
+      mapRef.current.removeControl(geolocateCtrl);
+      mapRef.current.off('move', updateCoordinates);
+      mapRef.current.remove();
+    };
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
+
+  useEffect(() => {
+    if (geometry.length !== 0) {
+      drawFeatureCollection();
+    }
+  }, [geometry]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  return (
+    <>
+      <div className="sidebar">
+        <div className="legend">
+          <p>Center Lat: {lat}</p>
+          <p>Center Long: {lng}</p>
+          <p>Current Zoom: {zoom}</p>
+        </div>
+        <div className="heading">
+          <h2>Attractions Nearby:</h2>
+        </div>
+
+        <div className="listings">
+          {places.map((place, index) => (
+            <div
+              key={index}
+              id={`listing-${place.properties.id}`}
+              className={`item ${index === 0 && 'active'}`}
+            >
+              <a href="#" className="title" onClick={() => handleClick(place)}>
+                {place.properties.title}
+              </a>
+              <div>
+                {place.properties.distance.toFixed(2)} {units} away
+              </div>
+            </div>
+          ))}
+        </div>
+      </div>
+      <div ref={mapContainerRef} className="map-container" />
+    </>
+  );
+}
+
+export default App;
+```
+
+  - To understand how the app works, let's break down this monster component's core functionality: 
```

Comment:
Looking great! Could you take these sections explaining the code and refactor so that they are brief comments in the code itself? I think this will be more intuitive for developers reading the passage